### DISCUSSION TITLE
QueryBuilder getRootAliases too wide return type?

### DISCUSSION THREAD

**Discussion by norkunas** [doctrine/doctrine2#9012] [Support Questions] [answered]
https://github.com/doctrine/orm/blob/0ae2a2ded8666ff33e5d09f800dbb4b9edae2d60/lib/Doctrine/ORM/QueryBuilder.php#L428-L456

I do not understand how `getRootAliases` could return anything other than string if `From::getAlias` always returns string?

**Comment by derrabus** [comment, marked as answer]
> I do not understand how `getRootAliases` could return anything other than string

Neither do I. 🙂
